File: CS50/SET_2/Legibilidade.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int grading(char text[500]);

int main(void)
{
    char text[500];
    printf("Digite seu texto: ");
    fgets(text, sizeof(text), stdin);

    int result = grading(text);
    if (result < 1)
    {
        printf("Before Grade 1");
    }
    else if (result > 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %i", result);
    }
}

int grading(char text[500])
{
    int words = 0;
    int characters = 0;
    int phrases = 0;

    int ignore_space = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (islower(text[i]) != 0 || isupper(text[i]) != 0)
        {
            characters++;
        }
        if (text[i] == ' ' && ignore_space == 0)
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            ignore_space = 1;
            phrases++;
            words++;
        }
        else if (ignore_space == 1)
        {
            ignore_space = 0;
        }
    }

    float L = (characters / (float) words) * 100;
    float S = ((float) phrases / words) * 100;

    float average_grading = 0.0588 * L - 0.296 * S - 15.8;
    return round(average_grading);
}

File: CS50/SET_6/legibilidade.py
def gradding_text(text):
    letters = sum(1 for char in text if char.isalpha())
    words = len(text.split())
    sentences = sum(1 for char in text if char in '.!?')
    
    L = (letters / words) * 100
    S = (sentences / words) * 100
    
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)

result = gradding_text(input("Text: "))
print(f"Grade {result}" if 1 <= result < 16 else "Before Grade 1" if result < 1 else "Grade 16+")